pkg/collector/corechecks/cluster/helm/helm.go

```diff
@@ -175,42 +184,104 @@ func helmTags(release *release, storageDriver helmStorage) []string {
 	return tags
 }
 
-func (hc *HelmCheck) releasesFromSecrets() ([]*release, error) {
-	secrets, err := hc.secretLister.List(helmSelector)
+func (hc *HelmCheck) addSecret(obj interface{}) {
+	secret, ok := obj.(*v1.Secret)
+	if !ok {
+		log.Warnf("Expected secret, got: %v", obj)
+		return
+	}
+
+	if !isManagedByHelm(secret) {
+		return
+	}
+
+	decodedRelease, err := decodeRelease(string(secret.Data["release"]))
 	if err != nil {
-		return nil, err
+		log.Warnf("error while decoding Helm release: %s", err)
+		return
 	}
 
-	var releases []*release
+	hc.releasesMut.Lock()
+	defer hc.releasesMut.Unlock()
+	hc.releases[k8sSecrets][decodedRelease.ID()] = decodedRelease
+}
 
-	for _, secret := range secrets {
-		deployedRelease, err := decodeRelease(string(secret.Data["release"]))
-		if err != nil {
-			return nil, fmt.Errorf("error while decoding Helm release: %s", err)
-		}
-		releases = append(releases, deployedRelease)
+func (hc *HelmCheck) deleteSecret(obj interface{}) {
+	secret, ok := obj.(*v1.Secret)
+	if !ok {
+		log.Warnf("Expected secret, got: %v", obj)
+		return
+	}
+
+	if !isManagedByHelm(secret) {
+		return
 	}
 
-	return releases, nil
+	decodedRelease, err := decodeRelease(string(secret.Data["release"]))
+	if err != nil {
+		log.Warnf("error while decoding Helm release: %s", err)
+		return
+	}
+
+	hc.releasesMut.Lock()
+	defer hc.releasesMut.Unlock()
+	delete(hc.releases[k8sSecrets], decodedRelease.ID())
 }
 
-func (hc *HelmCheck) releasesFromConfigMaps() ([]*release, error) {
-	configMaps, err := hc.configmapLister.List(helmSelector)
+func (hc *HelmCheck) updateSecret(_, obj interface{}) {
+	hc.addSecret(obj)
+}
+
+func (hc *HelmCheck) addConfigmap(obj interface{}) {
+	configmap, ok := obj.(*v1.ConfigMap)
+	if !ok {
+		log.Warnf("Expected configmap, got: %v", obj)
+		return
+	}
+
+	if !isManagedByHelm(configmap) {
+		return
+	}
+
+	decodedRelease, err := decodeRelease(configmap.Data["release"])
 	if err != nil {
-		return nil, err
+		log.Warnf("error while decoding Helm release: %s", err)
+		return
 	}
 
-	var releases []*release
+	hc.releasesMut.Lock()
+	defer hc.releasesMut.Unlock()
+	hc.releases[k8sConfigmaps][decodedRelease.ID()] = decodedRelease
```

Comment:
you might be able to wrap this 3 lignes in a function. I see the same 3 lignes 4t imes